src/CommonMark/Extensions/Link/LinkRenderer.php

```diff
@@ -95,6 +95,31 @@ public function getXmlAttributes(Node $node): array
         ];
     }
 
+    private function getNodeUrl(Node $node): string
+    {
+        $url = $node->getUrl();
+
+        $path = parse_url($url, PHP_URL_PATH);
+
+        if ($this->pathIsADomain($path)) {
+            return 'http://'.$path;
```

Comment:
```suggestion
            return 'https://'.$path;
```
wouldn't https be better? 🤔

Reply:
@ItsANameToo I used http because not all sites use https but all use http

Reply:
If a site doesn't use a secure setup I would say that's on them. By default we should redirect the user to the one with an SSL certificate and if that doesn't work then the user should go to the insecure one on their own.